modules/ocf_mail/files/spam/logging/examine-mail-log

```diff
@@ -115,14 +115,30 @@ def sending_too_much_mail(entry):
         return False  # no uid, so can't flag them
 
 
+def vhost_sending_too_much_mail(entry):
+    """Returns whether the given vhost address has sent out too many emails.
+
+    Args:
+        entry: a (address, count) tuple
+    """
+    addr, count = entry
+    return count > MAIL_PER_DAY
+
+
 if __name__ == '__main__':
     problems = defaultdict(lambda: defaultdict(list))
-    mail_count = defaultdict(int)
+    mail_count = Counter()
+    vhost_mail_count = Counter()
 
     for mail in map(json.loads, sys.stdin):
+        from_name, from_addr = mail['from']
+        from_domain = extract_domain(from_addr)
+
         uid = mail.get('uid')
 
-        if uid:
+        if is_vhost_domain(from_domain):
+            vhost_mail_count[from_addr] += 1
```

Comment:
ideally it would only apply this to mail actually sent by vhost users, as-is doesn't this also apply to mail sent e.g. via PHP?

i'd be worried about spam that chooses a random from address on the vhost for each message which would never get flagged here